src/Swagger/Serializer/ApiGatewayNormalizer.php

```diff
@@ -94,18 +96,18 @@ public function normalize($object, $format = null, array $context = [])
                     unset($data['definitions'][$definition]['properties'][$property]['readOnly']);
                 }
                 if (isset($propertyOptions['$ref']) && !preg_match('/^#\/definitions\/[A-z]+$/', $propertyOptions['$ref'])) {
-                    $data['definitions'][$definition]['properties'][$property]['$ref'] = str_replace(['-', '_'], '', $propertyOptions['$ref']);
+                    $data['definitions'][$definition]['properties'][$property]['$ref'] = str_replace(['-', ':', '_'], '', $propertyOptions['$ref']);
                 }
                 if (isset($propertyOptions['items']['$ref']) && !preg_match('/^#\/definitions\/[A-z]+$/', $propertyOptions['items']['$ref'])) {
-                    $data['definitions'][$definition]['properties'][$property]['items']['$ref'] = str_replace(['-', '_'], '', $propertyOptions['items']['$ref']);
+                    $data['definitions'][$definition]['properties'][$property]['items']['$ref'] = str_replace(['-', ':', '_'], '', $propertyOptions['items']['$ref']);
                 }
             }
         }
 
         // $data['definitions'] is an instance of \ArrayObject
         foreach (array_keys($data['definitions']->getArrayCopy()) as $definition) {
             if (!preg_match('/^[A-z]+$/', (string) $definition)) {
-                $data['definitions'][str_replace(['-', '_'], '', (string) $definition)] = $data['definitions'][$definition];
+                $data['definitions'][preg_replace('/[^\da-z]/i', '', (string) $definition)] = $data['definitions'][$definition];
```

Comment:
Anyway, let's just use: `/[^\dA-z]/`

Reply:
A-z represents char code 65 to 122.

![image](https://user-images.githubusercontent.com/46711821/67490572-baf80880-f673-11e9-9099-3de5e1f92688.png)

Reply:
Yeah, my bad, it should be `A-Za-z`.